site/app/models/gradeable/AutoGradedVersion.php

```diff
@@ -596,6 +596,58 @@ private function setSubmissionTimeInternal($submission_time) {
         }
     }
 
+    public function getMetrics_runtime() {
```

Comment:
This function (and all of the following functions) should have a return type annotation.